keps/sig-scheduling/2044-topology-awareness-in-kube-scheduler/README.md

```diff
@@ -0,0 +1,767 @@
+
+* Integration Tests
+   *  Default configuration (this plugin is disabled)
+     * no side effect on basic scheduling flow (and performance)
+
+   *  Enable this plugin
+     * basic workflow of this feature works (decision by scheduler is admitted by kubelet)
+     * basic negative path of this feature works (decision by scheduler is rejected by kubelet)
+
+* End-to-end tests
+
+Integration and End-to-end would Implementation of it does not constitute a difficulty, but requires appropriate multi-numa hardware for comprehensive testing of this feature. Comprehensive E2E testing of this would be done in order to graduate this feature from Alpha to Beta.
+
+
+
+
+### Graduation Criteria
+
+#### Alpha
+
+- [ ] Introducing `NodeResourceTopology` API to faciliatate communication between kubelet and kube-scheduler 
+- [ ] A new scheduler plugin `NodeResourceTopologyMatch` that makes topology-aware placement decisions
+    - [ ] Implementation of Filter extension point
+- [ ] Kubelet changes to populate `NodeResourceTopology` 
+- [ ] Unit tests and integration tests from [Test plans](#test-plans).
+
+#### Beta
+ 
+- [ ] Implementation of Score extension point
+- [ ] Add node E2E tests.
+- [ ] Provide beta-level documentation.
+
+#### GA
+
+- Add Conformance Tests
+- More rigorous testing—e.g., downgrade tests and scalability tests
+
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+No changes are required on upgrade to maintain previous behaviour.
+
+It is possible to downgrade kubelet on a node that was is using this capability by simply
+disabling the `ExposeNodeResourceTopology` feature gate and disabling the `NodeResourceTopologyMatch`
+scheduler plugin in case this plugin was enabled in the KubeScheduler configuration.   
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+Feature flag will apply to kubelet only, so version skew strategy is N/A.
```

Comment:
what happens in the kubelet when you disable the flag? Does the `NodeResourceTopology` object get deleted?
If I downgrade a kubelet, note that the object might stay there.